Update syntax to Python3.

adarnimrod opened the pull request:
Used 2to3. Update shebang to python3 as well. I ran it and it performed the backup without an issues.

Commit:
Update syntax to Python3.

Used 2to3. Update shebang to python3 as well.

dbnicholson reviewed (commented):
This is great. For a big change like this, I think we should send it upstream to @sgnn7 and then merge it back. I can do the review their, too.

Have you run it with python3?

dbnicholson commented:
>> I ran it and it performed the backup without an issues.

> Have you run it with python3?

Maybe I should read...

dbnicholson commented:
Since it looks like @sgnn7 would like to review a bit later in sgnn7/gh_fetch#17, I'll just review this now and we can rebase/merge later.

dbnicholson reviewed (commented):
All looks reasonable to me at a glance. The one thing I'd be worried about is handling of returned data from `urlopen()` and `Popen()`. We're treating them as strings, but AFAIK both interface return bytes unless you're specific about what you want. This is probably also influenced by the running locale. It would be interesting if you could test with `LC_ALL=C` and see if it still works. I expect we need a splattering of `.decode('utf-8')` in there. I believe `Popen()` has an `encoding` keyword argument that instructs it to open the streams in text mode.

Commit:
Have Popen output as text streams.

When passing universal_newlines=True to Popen stdout and stderr are text
streams instead of bytes. Supported since Python 3.2.

Commit:
Decode responses.

Responses from urlopen are bytes. Decode them using the default locale
encoding. Supported since Python 2.3.

Commit:
Retain some support for Python2.

Allow for running under Python 2.6 or 2.7. The invocation needs to be
`python update_repos`.

dbnicholson commented on update_repos:
```diff
@@ -88,10 +91,11 @@ def read_api_uri(uri, config, page=1):
     req = Request(uri, headers=headers)
-    return urlopen(req).read()
+    encoding = locale.getpreferredencoding()
+    return urlopen(req).read().decode(encoding)
```
I think if you don't pass an argument to `decode` it does this automatically, but it didn't hurt. The other potential fix would be to get the encoding once globally during initialization since it's not going to change during the lifetime of the process. This is fine, though.

adarnimrod replied:
Changed it to be a global variable.

dbnicholson reviewed (approved):
This is good enough to move forward with. We can always fix bugs after we see some live runs. Feel free to merge when you're ready. The existing job isn't running anymore since the server hosting that VM went down.

adarnimrod commented:
I ran a few more tests with Python 2 and 3, with UTF-8 and C locales (Python 2 still fails with the locale set to C). Anyway, merging.
